lib/Widgets/IncomingTransmissionListEntry.vala

```diff
@@ -199,13 +205,21 @@ public class Drop.Widgets.IncomingTransmissionListEntry : TransmissionListEntry
         set_progress_visible (false);
     }
 
+    private void swap_buttons () {
+        action_area.remove (accept_button);
+        action_area.remove (reject_button);
+        action_area.add (cancel_button);
+    }
+
     private void display_single_file (IncomingFileRequest file) {
-        set_primary_label (file.name);
+        set_primary_label (_("Incoming file from %s").printf(transmission.get_client_name ()));
+
+        file_string = file.name;
 
-        string info_text = format_size (file.size);
+        string info_text = "";
```

Comment:
true, you're right. Then maybe add some other text here for the case the one method fails or simply display the file size like before?